xchange-stream-dydx/src/main/java/info/bitrich/xchangestream/dydx/dydxStreamingMarketDataService.java

```diff
@@ -0,0 +1,81 @@
+package info.bitrich.xchangestream.dydx;
+
+import info.bitrich.xchangestream.core.StreamingMarketDataService;
+import info.bitrich.xchangestream.dydx.dto.v3.dydxInitialOrderBookMessage;
+import info.bitrich.xchangestream.dydx.dto.v3.dydxUpdateOrderBookMessage;
+import io.reactivex.Observable;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.marketdata.Ticker;
+import org.knowm.xchange.dto.marketdata.Trade;
+import org.knowm.xchange.exceptions.NotYetImplementedForExchangeException;
+
+import java.math.BigDecimal;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+/** Author: Max Gao (gaamox@tutanota.com) Created: 20-02-2021 */
+public class dydxStreamingMarketDataService implements StreamingMarketDataService {
+
+  private final dydxStreamingService service;
+
+  private final Map<CurrencyPair, SortedMap<BigDecimal, BigDecimal>> bids =
+      new ConcurrentHashMap<>();
+  private final Map<CurrencyPair, SortedMap<BigDecimal, BigDecimal>> asks =
+      new ConcurrentHashMap<>();
+
+  dydxStreamingMarketDataService(dydxStreamingService service) {
+    this.service = service;
+  }
+
+  private boolean containsPair(List<CurrencyPair> pairs, CurrencyPair pair) {
+    return pairs.stream().anyMatch(p -> p.equals(pair));
+  }
+
+  @Override
+  public Observable<OrderBook> getOrderBook(CurrencyPair currencyPair, Object... args) {
+    if (!containsPair(service.getProduct().getOrderBook(), currencyPair))
+      throw new UnsupportedOperationException(
+          String.format("The currency pair %s is not subscribed for orderbook", currencyPair));
+
+    final int maxDepth =
+        (args.length > 0 && args[0] instanceof Number) ? ((Number) args[0]).intValue() : 100;
+
+    return service
+        .getRawWebsocketTransactions(currencyPair, service.V3_ORDERBOOK)
```

Comment:
since you default to V3_ORDERBOOK and never adjust based on the v1,v3 of the API does this imply v1 wouldn't actually work here?

Reply:
I'll add some logic to support v1's orderbooks (margin and spot) in a follow-up commit.